src/modules/storagefacility/TableComponent.js

```diff
@@ -102,6 +104,70 @@ export class StorageUnitsContainer extends React.Component {
     });
   }
 
+  componentDidMount() {
+    this.scanner = subscribeToScanner((barCode) => {
+      this.props.clear();
+      const museumId = this.props.appSession.getMuseumId();
+      const collectionId = this.props.appSession.getCollectionId();
+      const token = this.props.appSession.getAccessToken();
+      if (barCode.uuid) {
+        this.props.findByUUID({uuid: barCode.code, museumId, token}).do((response) => {
+          if (!response) {
+            this.props.emitError({message: I18n.t('musit.errorMainMessages.scanner.noMatchingNode', {uuid: barCode.code})});
+          } else {
+            const isMoveHistoryActive = document.getElementsByClassName('moveHistory').length > 0;
+            const isMoveDialogActive = document.getElementsByClassName('moveDialog').length > 0;
+            if (isMoveDialogActive) {
+              this.props.updateMoveDialog(response, museumId, token);
+            } else if (isMoveHistoryActive) {
+              this.props.emitError({message: I18n.t('musit.errorMainMessages.scanner.cannotActOnObject')});
+            } else {
+              hashHistory.push('/magasin/' + response.id);
+            }
+          }
+        }).toPromise();
+      } else {
+        this.props.findByBarcode({barcode: barCode.code, museumId, collectionId, token}).do(response =>  {
+          if (!response) {
+            this.props.emitError({message: I18n.t('musit.errorMainMessages.scanner.noMatchingNodeOrObject', {barcode: barCode.code})});
+          } else if(Array.isArray(response)) {
+            if (response.length === 1) {
+              if (!response[0].currentLocationId) {
+                this.props.emitError({ message: I18n.t('musit.errorMainMessages.scanner.noCurrentLocation')});
+              } else {
+                const isMoveHistoryActive = document.getElementsByClassName('moveHistory').length > 0;
+                const isMoveDialogActive = document.getElementsByClassName('moveDialog').length > 0;
+                if (isMoveDialogActive) {
+                  this.props.updateMoveDialog(response[0], museumId, token);
+                } else if (isMoveHistoryActive) {
+                  this.props.emitError({message: I18n.t('musit.errorMainMessages.scanner.cannotActOnObject')});
+                } else {
+                  hashHistory.push('/magasin/' + response[0].currentLocationId + '/objects');
+                }
+              }
+            } else {
+              this.props.emitError({message: I18n.t('musit.errorMainMessages.scanner.noMatchingNodeOrObject', {barcode: barCode.code})});
+            }
+          } else if (response.nodeId) {
+            const isMoveHistoryActive = document.getElementsByClassName('moveHistory').length > 0;
+            const isMoveDialogActive = document.getElementsByClassName('moveDialog').length > 0;
```

Comment:
and here:
This feels like a code smell! Is it easy to send in this state so we don't need to interact with the DOM.